### DISCUSSION TITLE
Can install report downloaded packages?

### DISCUSSION THREAD

**Discussion by Fish-in-a-Barrel** [pnpm/pnpm#3891] [Q&A]
In my CI environment, I'm seeing a higher download count than I expect, and I'm trying to figure out how to investigate that further.

When running `pnpm install`, is there a way to have it list the packages that it had to download versus those it linked from the cache?

**Comment by zkochan** [comment]
```
pnpm install --reporter ndjson
```

**Reply by Fish-in-a-Barrel** [reply]
Well, that's certainly a complete report.

Any idea how to find the 20 things that were downloaded in this 10k line file? I've searched for 'download', but that's not found. I tried filtering out everything that was a successful cache hit, but that's not leaving anything. So either I'm not able to find the right message, it's not logging the download, or the "downloaded" count doesn't mean what I think it does.

**Reply by zkochan** [reply]
When package is downloaded, a "fetched" event is logged

When a package is found in the store, it will be "found_in_store"

https://github.com/pnpm/pnpm/blob/930e104da3c7a1da8f0096c9e0d1bb323bcd0735/packages/default-reporter/src/reporterForClient/reportProgress.ts#L133-L138

**Reply by Fish-in-a-Barrel** [reply]
That's odd then. Whenever my pipeline runs, the standard reporter counts 20 downloaded packages, but there are zero "fetched" log entries in the ndjson log.
